doc/design.md

```diff
@@ -0,0 +1,94 @@
+# Design
+
+## DPPP: a framework for streaming radio interferometric pipelines
+DPPP (DP3, short for Default PreProcessing Pipeline) is a framework
+for efficiently processing time-ordered visibilities,
+e.g. freshly correlated data. It was developed for the LOFAR
+telescope, and is used for preprocessing its imaging pipelines.
+DPPP user documentation is found at the [LOFAR Wiki](https://goo.gl/2uAaN9).
+
+When DPPP reads a chunk of data, it calls all the steps to do
+all processing possible on this data. The last step will then
+write out the processed data. In this way, the data is read
+and written only once.
+
+![High level overview of DPPP](doc/images/diagram.png)
+
+DPPP runs a pipeline in clearly defined stages, to make spotting
+errors as easy as possible.
+* **Initialise** This connects all the processing steps together. This step is data independent.
+* **Tune At** this stage only metadata, such as the number of channels, is read, and steps are adjusted to this.
+* **Process** Data is piped through the steps in buffers of time. The buffer size is defined by the steps.
+
+## Measurement Set standard
+DPPP adheres to the Measurement Set (MS) standard, so it
+can be used with data from all radio telescopes. DPPP can
+also create a new MS. When an appropriate input step is in
+place (e.g. a streaming connection to a correlator), MSs can
+be written from scratch.
+
+DPPP supports only regularly shaped MSs, meaning that every
+time slot must have the same number of baselines and antennas.
+Only one spectral window can be used.
+
+## Built-in steps
+DPPP contains built-in steps that are linked together and process one time slot at a time and is therefore suitable for large measurement sets.
+All the built-in steps derive from the [DPStep](@ref DP3::DPPP::DPStep) class.
+
+[DPRun](@ref DP3::DPPP::DPRun) initialises and orchestrates the configured steps.
+It begins with setting the required info for all the first step and its next step(s).
+Next, each buffer (containing a single time slot) is processed by the first step.
+When processed, it invokes the process function of the next step.
+Once all buffers are processed it finishes the processing of the first step and subsequent steps.
+It wraps it with adding some data to the MeasurementSet written/updated which is started at the last step.
+The figure below gives a graphical overview of the [DPStep](@ref DP3::DPPP::DPStep) flow.
+
+![Process flow of DPPP](doc/images/flow.png)
+
+### Calibration
+The DPPP step [GainCal](@ref DP3::DPPP::GainCal) implements many variants of direction
+independent calibration. It uses StefCal, with many
+extra features, such as fitting a function over frequency.
+Also full-Jones calibration is supported. Results are written
+to ParmDB; H5Parm support is in development.
+
+### AOFlagger
+A shallow wrapper exists that calls AOFlagger from DPPP.
```

Comment:
It looks like the shallow wrapper is in fact a normal DPStep class, which is worth mentioning here, like:
"DPPP contains a DPStep class which a shallow wrapper that calls AOFlagger."